{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "setwd(\"/Users/rebecca/sudmant/analyses/myotis/analysis/exploratory/species_peaks\")\n",
    "\n",
    "library(scales)\n",
    "library(dplyr)\n",
    "library(ggplot2)\n",
    "library(data.table)\n",
    "library(GenomicRanges)\n",
    "\n",
    "\n",
    "myo_meta <- read.csv(\"/Users/rebecca/sudmant/analyses/myotis/data/myotis_meta.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "## Which TEs are in accessible vs. inaccessible regions?\n",
    "\n",
    "frac_list <- lapply(seq_along(myo_meta$Abbr), function(i) {\n",
    "\n",
    "  ## Load in TE peak data\n",
    "  features_peaks <- fread(\n",
    "    file.path(\"results/data\", paste0(myo_meta$Abbr[i], \"_TEs_peaks.tsv\")), \n",
    "    data.table = FALSE\n",
    "  )\n",
    "  \n",
    "  features_peaks$Type <- features_peaks$V7\n",
    "  features_peaks$Type[grep(\"^DNA\", features_peaks$Type)] <- \"DNA\"\n",
    "  features_peaks$Type[grep(\"^SINE\", features_peaks$Type)] <- \"SINE\"\n",
    "  features_peaks$Type[grep(\"^LINE\", features_peaks$Type)] <- \"LINE\"\n",
    "  features_peaks$Type[grep(\"^LTR\", features_peaks$Type)] <- \"LTR\"\n",
    "  features_peaks$Type[grep(\"^Simple\", features_peaks$Type)] <- \"Simple repeat\"\n",
    "  features_peaks$Type[grep(\"Unknown\", features_peaks$Type)] <- \"Unknown\"\n",
    "  features_peaks$Type[grep(\"RNA\", features_peaks$Type)] <- \"rRNA/tRNA/snRNA\"\n",
    "  features_peaks$Type[grep(\"^Sat\", features_peaks$Type)] <- \"Other\"\n",
    "  features_peaks$Type[grep(\"^Retro\", features_peaks$Type)] <- \"Other\"\n",
    "  features_peaks$Type <- gsub(\"_\", \" \", features_peaks$Type)\n",
    "                          \n",
    "  ## Which TEs are occupied by peaks?\n",
    "  df_all <- features_peaks %>%\n",
    "    dplyr::group_by(ID = paste(V1, V2, V3)) %>% \n",
    "    dplyr::reframe(\n",
    "      Peak_ID = paste(unique(V11), collapse = \", \"),\n",
    "      Type = paste(unique(Type), collapse = \", \"),\n",
    "    ) %>%\n",
    "    dplyr::mutate(\n",
    "      Peak = ifelse(Peak_ID == \".\", FALSE, TRUE),\n",
    "      Type = ifelse(grepl(\",\", Type), \"Unknown\", Type)\n",
    "    ) %>%\n",
    "    dplyr::group_by(Type, Peak) %>%\n",
    "    dplyr::reframe(No.Elements = n())\n",
    "    \n",
    "  df_DNA <- features_peaks %>%\n",
    "    dplyr::filter(grepl(\"DNA\", V7)) %>%\n",
    "    dplyr::group_by(ID = paste(V1, V2, V3)) %>% \n",
    "    dplyr::reframe(\n",
    "      Peak_ID = paste(unique(V11), collapse = \", \"),\n",
    "      Type = paste(\n",
    "        gsub(\"DNA/\", \"\", unique(V7), fixed = TRUE), \n",
    "        collapse = \", \"\n",
    "      )\n",
    "    ) %>%\n",
    "    dplyr::mutate(\n",
    "      Peak = ifelse(Peak_ID == \".\", FALSE, TRUE),\n",
    "      Type = ifelse(grepl(\",\", Type), \"Unknown\", Type)\n",
    "    ) %>%\n",
    "    dplyr::group_by(Type, Peak) %>%\n",
    "    dplyr::reframe(No.Elements = n())\n",
    "    \n",
    "  return(list(All = df_all, DNA = df_DNA))\n",
    "  \n",
    "})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "## Plot TE divergence in accessible vs. nonaccessible regions:\n",
    "\n",
    "feature_files <- list.files(path = \"results/data\", \n",
    "                            pattern = \"TEs_peaks.tsv\", \n",
    "                            full.names = TRUE)\n",
    "\n",
    "pdf(\"results/figures/peak_status_vs_TE_divergence.pdf\", width = 6, height = 7)\n",
    "\n",
    "lapply(seq_along(myo_meta$Abbr), function(i) {\n",
    "  \n",
    "  ## Load in feature data with intersecting peaks (file contains all features for a given species, irrespective of whether they intersected a peak)\n",
    "  features_peaks <- fread(\n",
    "    feature_files[grep(myo_meta$Abbr[i], feature_files)], \n",
    "    data.table = FALSE\n",
    "  )\n",
    "  \n",
    "  ## Group by feature, and collapse all peaks associated with that feature\n",
    "  df <- features_peaks %>%\n",
    "    dplyr::mutate(ID = paste(V1, V2, V3, V7)) %>% \n",
    "    dplyr::group_by(ID) %>%\n",
    "    dplyr::reframe(\n",
    "      Peak_ID = paste(unique(V11), collapse = \", \"),\n",
    "      Type = unique(V7),\n",
    "      Repeat_Name = unique(V6),\n",
    "      Divergence = unique(V5) \n",
    "    )\n",
    "    \n",
    "  df$Peak <- \"No peak\"\n",
    "  df$Peak[df$Peak_ID != \".\"] <- \"Peak\"\n",
    "  df$Peak <- factor(df$Peak, levels = c(\"Peak\", \"No peak\"))\n",
    "    \n",
    "  print(\n",
    "    ggplot(df, aes(x = Peak, y = Divergence)) +\n",
    "      geom_violin(color = \"grey\", fill = \"grey\") +\n",
    "      geom_boxplot(notch = TRUE, width = .3, outlier.shape = NA) +\n",
    "      theme_minimal() +\n",
    "      theme(axis.title.x = element_text(margin = margin(t = 10)),\n",
    "            axis.text.x = element_text(),\n",
    "            axis.title.y = element_text(margin = margin(r = 15)),\n",
    "            plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "      labs(title = myo_meta$Field_Name[i], \n",
    "            subtitle = \"Transposable element divergence\") +\n",
    "      xlab(\"TE status\") + ylab(\"Divergence\") +\n",
    "      scale_y_continuous(labels = comma) \n",
    "  )\n",
    "  \n",
    "  print(\n",
    "    ggplot(subset(df, grepl(\"DNA\", Type)), aes(x = Peak, y = Divergence)) +\n",
    "      geom_violin(color = \"grey\", fill = \"grey\") +\n",
    "      geom_boxplot(notch = TRUE, width = .3, outlier.shape = NA) +\n",
    "      theme_minimal() +\n",
    "      theme(axis.title.x = element_text(margin = margin(t = 10)),\n",
    "            axis.text.x = element_text(),\n",
    "            axis.title.y = element_text(margin = margin(r = 15)),\n",
    "            plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "      labs(title = myo_meta$Field_Name[i], \n",
    "            subtitle = \"DNA transposable element divergence\") +\n",
    "      xlab(\"TE status\") + ylab(\"Divergence\") +\n",
    "      scale_y_continuous(labels = comma) \n",
    "  )\n",
    "  \n",
    "})\n",
    "       \n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "## Plot proportion of peaks with TEs\n",
    "\n",
    "frac_peaks_list <- lapply(seq_along(myo_meta$Abbr), function(i) {\n",
    "  peak_features <- fread(\n",
    "    peak_files[grep(myo_meta$Abbr[i], peak_files)], data.table = FALSE\n",
    "  )\n",
    "  uniq_peaks <- peak_features %>%\n",
    "    dplyr::mutate(TE = ifelse(V10 == \".\", FALSE, TRUE)) %>%\n",
    "    dplyr::group_by(V4) %>%\n",
    "    dplyr::reframe(TE = unique(TE))\n",
    "    \n",
    "  return(sum(uniq_peaks$TE == TRUE) / nrow(uniq_peaks))\n",
    "})\n",
    "\n",
    "df <- data.frame(Species = myo_meta$Field_Name, \n",
    "                 Proportion = do.call(rbind, frac_peaks_list), \n",
    "                 row.names = NULL) %>%\n",
    "  dplyr::arrange(desc(Proportion))\n",
    "\n",
    "df$Species <- factor(df$Species, levels = df$Species)\n",
    "\n",
    "pdf(\"results/figures/peaks_TE_proportion.pdf\", width = 8, height = 6)\n",
    "\n",
    "print(\n",
    "  ggplot(df, aes(x = Species, y = Proportion)) +\n",
    "    geom_bar(stat = \"identity\") +\n",
    "    theme_minimal() +\n",
    "    theme(axis.title.x = element_blank(),\n",
    "          axis.text.x = element_text(angle = 45, hjust = 1),\n",
    "          axis.title.y = element_text(margin = margin(r = 15)),\n",
    "          plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "    labs(title = \"Proportion of peaks in transposable elements\") +\n",
    "    ylab(\"Proportion\")\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# ## Plot fraction of genome comprised of TEs vs. TEs covered by peaks\n",
    "\n",
    "# frac_TEs_list <- lapply(seq_along(myo_meta$Abbr), function(i) {\n",
    "\n",
    "#   ## Load in TE GTF\n",
    "#   spec_TEs <- fread(file.path(\"/Users/rebecca/sudmant/analyses/myotis/data/genomes/bat_genomes/repeatMasker\", \n",
    "#                               paste0(myo_meta$Abbr[i], \"1_repetMasker.gff3\")), skip = 1, data.table = FALSE)\n",
    "\n",
    "#   ## Load in chromosome lengths\n",
    "#   chromsize <- fread(paste0(\"/Users/rebecca/sudmant/analyses/myotis/data/\", \n",
    "#                             myo_meta$Abbr[i], \"_chromsizes\"), \n",
    "#                      data.table = FALSE)\n",
    "\n",
    "#   ## Load in TE peak data\n",
    "#   features_peaks <- fread(\n",
    "#     file.path(\"results/data\", paste0(myo_meta$Abbr[i], \"_TEs_peaks.tsv\")), \n",
    "#     data.table = FALSE\n",
    "#   )\n",
    "                          \n",
    "#   ## Get fraction of genome occupied by TEs\n",
    "#   spec_TEs <- spec_TEs[,c(1, 4:5, 7)] \n",
    "#   colnames(spec_TEs) <- c(\"seqnames\", \"start\", \"end\", \"strand\")\n",
    "#   bps_TEs <- sum(width(GenomicRanges::reduce(\n",
    "#     makeGRangesFromDataFrame(spec_TEs, ignore.strand = TRUE)\n",
    "#   )))\n",
    "#   frac_genome <- bps_TEs/sum(chromsize[,2])\n",
    "\n",
    "#   ## Get fraction of TEs occupied by peaks:\n",
    "#   features_peaks <- features_peaks %>%\n",
    "#     dplyr::group_by(ID = paste(V1, V2, V3)) %>% \n",
    "#     dplyr::reframe(\n",
    "#       Peak_ID = paste(unique(V11), collapse = \", \")\n",
    "#     ) %>%\n",
    "#     dplyr::mutate(\n",
    "#       Peak = ifelse(Peak_ID == \".\", FALSE, TRUE)\n",
    "#     )\n",
    "#   frac_peaks <- sum(features_peaks$Peak)/nrow(features_peaks)       \n",
    "\n",
    "#   return(data.frame(Species = myo_meta$Abbr[i], \n",
    "#                     Frac_Genome = frac_genome,\n",
    "#                     Frac_Peaks = frac_peaks))\n",
    "  \n",
    "# }) \n",
    "\n",
    "# df <- do.call(rbind, frac_TEs_list)\n",
    "\n",
    "# pdf(\"results/figures/TE_peak_proportion.pdf\")\n",
    "\n",
    "# print(\n",
    "#   ggplot(df, aes(x = Frac_Genome, y = Frac_Peaks)) +\n",
    "#     geom_text(aes(label = Species)) +\n",
    "#     theme_minimal() +\n",
    "#     theme(axis.title.x = element_text(margin = margin(t = 15)),\n",
    "#           axis.text.x = element_text(),\n",
    "#           axis.title.y = element_text(margin = margin(r = 15)),\n",
    "#           plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "#     labs(title = \"Transposable elements\") +\n",
    "#     xlab(\"Fraction genome with TEs\") + ylab(\"Fraction TEs with peaks\") +\n",
    "#     scale_x_continuous(limits = c(min(df$Frac_Genome - .001), max(df$Frac_Genome + .001)))\n",
    "# )\n",
    "\n",
    "# dev.off()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
