{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: 'dplyr'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: 'data.table'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: 'BiocGenerics'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:stats':\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,\n",
      "    as.data.frame, basename, cbind, colnames, dirname, do.call,\n",
      "    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\n",
      "    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\n",
      "    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\n",
      "    tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: 'S4Vectors'\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:data.table':\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    first, rename\n",
      "\n",
      "\n",
      "The following object is masked from 'package:utils':\n",
      "\n",
      "    findMatches\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:base':\n",
      "\n",
      "    I, expand.grid, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "\n",
      "Attaching package: 'IRanges'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:data.table':\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following objects are masked from 'package:dplyr':\n",
      "\n",
      "    collapse, desc, slice\n",
      "\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n"
     ]
    }
   ],
   "source": [
    "setwd(\"/Users/rebecca/sudmant/analyses/myotis/analysis/exploratory/species_peaks\")\n",
    "\n",
    "library(dplyr)\n",
    "library(scales)\n",
    "library(ggplot2)\n",
    "library(data.table)\n",
    "library(GenomicRanges)\n",
    "\n",
    "myo_meta <- read.csv(\"/Users/rebecca/sudmant/analyses/myotis/data/myotis_meta.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "## Which TEs are in accessible vs. inaccessible regions?\n",
    "\n",
    "frac_TEs_list <- lapply(seq_along(myo_meta$Abbr), function(i) {\n",
    "\n",
    "  ## Load in TE peak data\n",
    "  features_peaks <- fread(\n",
    "    file.path(\"results/data\", paste0(myo_meta$Abbr[i], \"_TEs_peaks.tsv\")), \n",
    "    data.table = FALSE\n",
    "  )\n",
    "  \n",
    "  features_peaks$Type <- features_peaks$V7\n",
    "  features_peaks$Type[grep(\"^DNA\", features_peaks$Type)] <- \"DNA\"\n",
    "  features_peaks$Type[grep(\"^SINE\", features_peaks$Type)] <- \"SINE\"\n",
    "  features_peaks$Type[grep(\"^LINE\", features_peaks$Type)] <- \"LINE\"\n",
    "  features_peaks$Type[grep(\"^LTR\", features_peaks$Type)] <- \"LTR\"\n",
    "  features_peaks$Type[grep(\"^Simple\", features_peaks$Type)] <- \"Simple repeat\"\n",
    "  features_peaks$Type[grep(\"Unknown\", features_peaks$Type)] <- \"Unknown\"\n",
    "  features_peaks$Type[grep(\"RNA\", features_peaks$Type)] <- \"rRNA/tRNA/snRNA\"\n",
    "  features_peaks$Type[grep(\"^Sat\", features_peaks$Type)] <- \"Other\"\n",
    "  features_peaks$Type[grep(\"^Retro\", features_peaks$Type)] <- \"Other\"\n",
    "  features_peaks$Type <- gsub(\"_\", \" \", features_peaks$Type)\n",
    "                          \n",
    "  ## Which TEs are occupied by peaks?\n",
    "  df <- features_peaks %>%\n",
    "    dplyr::group_by(ID = paste(V1, V2, V3)) %>% \n",
    "    dplyr::reframe(\n",
    "      Peak_ID = paste(unique(V11), collapse = \", \"),\n",
    "      Type = paste(unique(Type), collapse = \", \"),\n",
    "      DNA_Type = paste(unique(V7), collapse = \", \")\n",
    "    ) %>%\n",
    "    dplyr::mutate(\n",
    "      Peak = ifelse(Peak_ID == \".\", FALSE, TRUE),\n",
    "      Type = ifelse(grepl(\",\", Type), \"Unknown\", Type),\n",
    "      DNA_Type = ifelse(grepl(\",\", DNA_Type), \"Unknown\", DNA_Type)\n",
    "    ) %>%\n",
    "    dplyr::group_by(Type, DNA_Type, Peak) %>%\n",
    "    dplyr::reframe(No.Elements = n()) %>%\n",
    "    dplyr::mutate(Species = myo_meta$Field_Name[i])\n",
    "    \n",
    "  return(df)\n",
    "  \n",
    "})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "## Proportion of TEs with peaks per species\n",
    "\n",
    "pdf(\"results/figures/TE_peak_proportion.pdf\", width = 13, height = 9)\n",
    "\n",
    "## Summarize over all TE types:\n",
    "\n",
    "df <- do.call(rbind, frac_TEs_list) %>%\n",
    "  dplyr::group_by(Species) %>%\n",
    "  dplyr::mutate(\n",
    "    Frac_TEs = No.Elements/sum(No.Elements)\n",
    "  )\n",
    "  \n",
    "x_order <- df %>%\n",
    "  dplyr::filter(Peak == TRUE) %>%\n",
    "  dplyr::group_by(Species) %>%\n",
    "  dplyr::reframe(n = sum(Frac_TEs)) %>%\n",
    "  dplyr::arrange(desc(n))\n",
    "  \n",
    "df$Species <- factor(df$Species, levels = x_order$Species)\n",
    "\n",
    "print(\n",
    "  ggplot(df, aes(x = Species, y = Frac_TEs, fill = Peak, color = Peak)) +\n",
    "    geom_bar(stat = \"identity\") +\n",
    "    theme_minimal() +\n",
    "    theme(axis.title.x = element_blank(),\n",
    "          axis.text.x = element_text(angle = 45, hjust = 1),\n",
    "          axis.title.y = element_text(margin = margin(r = 15)),\n",
    "          panel.grid = element_blank(),\n",
    "          plot.margin = unit(c(4, 6, 4, 6), \"cm\")) +\n",
    "    labs(title = \"Proportion of transposable elements with peaks\") +\n",
    "    ylab(\"Proportion\")\n",
    ")\n",
    "\n",
    "# ## Stratify by TE type:\n",
    "\n",
    "# print(\n",
    "#   ggplot(df, aes(x = Type, y = Frac_TEs, fill = Peak)) +\n",
    "#     geom_bar(stat = \"identity\") +\n",
    "#     theme_minimal() +\n",
    "#     theme(axis.title.x = element_blank(),\n",
    "#           axis.text.x = element_text(angle = 45, hjust = 1),\n",
    "#           axis.title.y = element_text(margin = margin(r = 15)),\n",
    "#           panel.grid = element_blank(),\n",
    "#           plot.margin = unit(c(1, 3, 1, 3), \"cm\")) +\n",
    "#     labs(title = \"Proportion of transposable elements with peaks\") +\n",
    "#     ylab(\"Proportion\") +\n",
    "#     facet_wrap(. ~ Species)\n",
    "# )\n",
    "\n",
    "# print(\n",
    "#   ggplot(df_all, aes(x = Type, y = No.Elements, fill = Peak)) +\n",
    "#     geom_bar(stat = \"identity\") +\n",
    "#     theme_minimal() +\n",
    "#     theme(axis.title.x = element_blank(),\n",
    "#           axis.text.x = element_text(angle = 45, hjust = 1),\n",
    "#           axis.title.y = element_text(margin = margin(r = 15)),\n",
    "#           panel.grid = element_blank(),\n",
    "#           plot.margin = unit(c(1, 3, 1, 3), \"cm\")) +\n",
    "#     labs(title = \"# of transposable elements with peaks\") +\n",
    "#     ylab(\"# elements\") + scale_y_continuous(labels = comma) +\n",
    "#     facet_wrap(. ~ Species)\n",
    "# )\n",
    "\n",
    "## Subset to DNA elements:\n",
    "\n",
    "# df_DNA <- df %>%\n",
    "#   dplyr::filter(grepl(\"DNA\", Type)) %>%\n",
    "#   dplyr::group_by(Species) %>%\n",
    "#   dplyr::mutate(\n",
    "#     Frac_TEs = No.Elements/sum(No.Elements)\n",
    "#   )\n",
    "  \n",
    "# print(\n",
    "#   ggplot(df_DNA, aes(x = DNA_Type, y = Frac_TEs, fill = Peak)) +\n",
    "#     geom_bar(stat = \"identity\") +\n",
    "#     theme_minimal() +\n",
    "#     theme(axis.title.x = element_blank(),\n",
    "#           axis.text.x = element_text(angle = 45, hjust = 1, size = 6),\n",
    "#           axis.title.y = element_text(margin = margin(r = 15)),\n",
    "#           panel.grid = element_blank(),\n",
    "#           plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "#     labs(title = \"Proportion of transposable elements with peaks\") +\n",
    "#     ylab(\"Proportion\") +\n",
    "#     facet_wrap(. ~ Species, scales = \"free_x\")\n",
    "# )\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "## Plot TE divergence in accessible vs. nonaccessible regions:\n",
    "\n",
    "feature_files <- list.files(path = \"results/data\", \n",
    "                            pattern = \"TEs_peaks.tsv\", \n",
    "                            full.names = TRUE)\n",
    "\n",
    "pdf(\"results/figures/peak_status_vs_TE_divergence.pdf\", width = 6, height = 7)\n",
    "\n",
    "lapply(seq_along(myo_meta$Abbr), function(i) {\n",
    "  \n",
    "  ## Load in feature data with intersecting peaks (file contains all features for a given species, irrespective of whether they intersected a peak)\n",
    "  features_peaks <- fread(\n",
    "    feature_files[grep(myo_meta$Abbr[i], feature_files)], \n",
    "    data.table = FALSE\n",
    "  )\n",
    "  \n",
    "  ## Group by feature, and collapse all peaks associated with that feature\n",
    "  df <- features_peaks %>%\n",
    "    dplyr::mutate(ID = paste(V1, V2, V3, V7)) %>% \n",
    "    dplyr::group_by(ID) %>%\n",
    "    dplyr::reframe(\n",
    "      Peak_ID = paste(unique(V11), collapse = \", \"),\n",
    "      Type = unique(V7),\n",
    "      Divergence = unique(V5) \n",
    "    ) %>%\n",
    "    dplyr::mutate(\n",
    "      Peak = ifelse(Peak_ID == \".\", \"No peak\", \"Peak\")\n",
    "    )\n",
    "\n",
    "  df$Peak <- factor(df$Peak, levels = c(\"Peak\", \"No peak\"))\n",
    "    \n",
    "  print(\n",
    "    ggplot(df, aes(x = Peak, y = Divergence)) +\n",
    "      geom_violin(color = \"grey\", fill = \"grey\") +\n",
    "      geom_boxplot(notch = TRUE, width = .3, outlier.shape = NA) +\n",
    "      theme_minimal() +\n",
    "      theme(axis.title.x = element_text(margin = margin(t = 10)),\n",
    "            axis.text.x = element_text(),\n",
    "            axis.title.y = element_text(margin = margin(r = 15)),\n",
    "            plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "      labs(title = myo_meta$Field_Name[i], \n",
    "            subtitle = \"Transposable element divergence\") +\n",
    "      xlab(\"TE status\") + ylab(\"Divergence\") +\n",
    "      scale_y_continuous(labels = comma) \n",
    "  )\n",
    "  \n",
    "  print(\n",
    "    ggplot(subset(df, grepl(\"DNA\", Type)), aes(x = Peak, y = Divergence)) +\n",
    "      geom_violin(color = \"grey\", fill = \"grey\") +\n",
    "      geom_boxplot(notch = TRUE, width = .3, outlier.shape = NA) +\n",
    "      theme_minimal() +\n",
    "      theme(axis.title.x = element_text(margin = margin(t = 10)),\n",
    "            axis.text.x = element_text(),\n",
    "            axis.title.y = element_text(margin = margin(r = 15)),\n",
    "            plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "      labs(title = myo_meta$Field_Name[i], \n",
    "            subtitle = \"DNA transposable element divergence\") +\n",
    "      xlab(\"TE status\") + ylab(\"Divergence\") +\n",
    "      scale_y_continuous(labels = comma) \n",
    "  )\n",
    "  \n",
    "})\n",
    "       \n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "i = 1\n",
    "\n",
    "feature_files <- list.files(path = \"results/data\", \n",
    "                            pattern = \"TEs_peaks.tsv\", \n",
    "                            full.names = TRUE)\n",
    "## Load in feature data with intersecting peaks (file contains all features for a given species, irrespective of whether they intersected a peak)\n",
    "  features_peaks <- fread(\n",
    "    feature_files[grep(myo_meta$Abbr[i], feature_files)], \n",
    "    data.table = FALSE\n",
    "  )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 x 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>V1</th><th scope=col>V2</th><th scope=col>V3</th><th scope=col>V4</th><th scope=col>V5</th><th scope=col>V6</th><th scope=col>V7</th><th scope=col>V8</th><th scope=col>V9</th><th scope=col>V10</th><th scope=col>V11</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>SUPER__1</td><td>    1</td><td>11309</td><td>12719</td><td> 0.6</td><td>(ACCCTA)n   </td><td>Simple_repeat</td><td>SUPER__1</td><td>9940</td><td>10930</td><td>Interval_1</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>SUPER__1</td><td>12795</td><td>12844</td><td>   37</td><td> 4.2</td><td>(CCCT)n     </td><td>Simple_repeat</td><td>.       </td><td>  -1</td><td>   -1</td><td>.         </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>SUPER__1</td><td>14915</td><td>14958</td><td>   13</td><td>15.2</td><td>(ACAGG)n    </td><td>Simple_repeat</td><td>.       </td><td>  -1</td><td>   -1</td><td>.         </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>SUPER__1</td><td>15280</td><td>15371</td><td>  365</td><td>18.7</td><td>MyoBra-1.148</td><td>LINE         </td><td>.       </td><td>  -1</td><td>   -1</td><td>.         </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>SUPER__1</td><td>15605</td><td>15663</td><td>   37</td><td>11.0</td><td>(AGGAGGG)n  </td><td>Simple_repeat</td><td>.       </td><td>  -1</td><td>   -1</td><td>.         </td></tr>\n",
       "\t<tr><th scope=row>6</th><td>SUPER__1</td><td>18026</td><td>18069</td><td>   13</td><td> 5.4</td><td>(ACACGC)n   </td><td>Simple_repeat</td><td>.       </td><td>  -1</td><td>   -1</td><td>.         </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 x 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & V1 & V2 & V3 & V4 & V5 & V6 & V7 & V8 & V9 & V10 & V11\\\\\n",
       "  & <chr> & <int> & <int> & <int> & <dbl> & <chr> & <chr> & <chr> & <int> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & SUPER\\_\\_1 &     1 & 11309 & 12719 &  0.6 & (ACCCTA)n    & Simple\\_repeat & SUPER\\_\\_1 & 9940 & 10930 & Interval\\_1\\\\\n",
       "\t2 & SUPER\\_\\_1 & 12795 & 12844 &    37 &  4.2 & (CCCT)n      & Simple\\_repeat & .        &   -1 &    -1 & .         \\\\\n",
       "\t3 & SUPER\\_\\_1 & 14915 & 14958 &    13 & 15.2 & (ACAGG)n     & Simple\\_repeat & .        &   -1 &    -1 & .         \\\\\n",
       "\t4 & SUPER\\_\\_1 & 15280 & 15371 &   365 & 18.7 & MyoBra-1.148 & LINE          & .        &   -1 &    -1 & .         \\\\\n",
       "\t5 & SUPER\\_\\_1 & 15605 & 15663 &    37 & 11.0 & (AGGAGGG)n   & Simple\\_repeat & .        &   -1 &    -1 & .         \\\\\n",
       "\t6 & SUPER\\_\\_1 & 18026 & 18069 &    13 &  5.4 & (ACACGC)n    & Simple\\_repeat & .        &   -1 &    -1 & .         \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 x 11\n",
       "\n",
       "| <!--/--> | V1 &lt;chr&gt; | V2 &lt;int&gt; | V3 &lt;int&gt; | V4 &lt;int&gt; | V5 &lt;dbl&gt; | V6 &lt;chr&gt; | V7 &lt;chr&gt; | V8 &lt;chr&gt; | V9 &lt;int&gt; | V10 &lt;int&gt; | V11 &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | SUPER__1 |     1 | 11309 | 12719 |  0.6 | (ACCCTA)n    | Simple_repeat | SUPER__1 | 9940 | 10930 | Interval_1 |\n",
       "| 2 | SUPER__1 | 12795 | 12844 |    37 |  4.2 | (CCCT)n      | Simple_repeat | .        |   -1 |    -1 | .          |\n",
       "| 3 | SUPER__1 | 14915 | 14958 |    13 | 15.2 | (ACAGG)n     | Simple_repeat | .        |   -1 |    -1 | .          |\n",
       "| 4 | SUPER__1 | 15280 | 15371 |   365 | 18.7 | MyoBra-1.148 | LINE          | .        |   -1 |    -1 | .          |\n",
       "| 5 | SUPER__1 | 15605 | 15663 |    37 | 11.0 | (AGGAGGG)n   | Simple_repeat | .        |   -1 |    -1 | .          |\n",
       "| 6 | SUPER__1 | 18026 | 18069 |    13 |  5.4 | (ACACGC)n    | Simple_repeat | .        |   -1 |    -1 | .          |\n",
       "\n"
      ],
      "text/plain": [
       "  V1       V2    V3    V4    V5   V6           V7            V8       V9  \n",
       "1 SUPER__1     1 11309 12719  0.6 (ACCCTA)n    Simple_repeat SUPER__1 9940\n",
       "2 SUPER__1 12795 12844    37  4.2 (CCCT)n      Simple_repeat .          -1\n",
       "3 SUPER__1 14915 14958    13 15.2 (ACAGG)n     Simple_repeat .          -1\n",
       "4 SUPER__1 15280 15371   365 18.7 MyoBra-1.148 LINE          .          -1\n",
       "5 SUPER__1 15605 15663    37 11.0 (AGGAGGG)n   Simple_repeat .          -1\n",
       "6 SUPER__1 18026 18069    13  5.4 (ACACGC)n    Simple_repeat .          -1\n",
       "  V10   V11       \n",
       "1 10930 Interval_1\n",
       "2    -1 .         \n",
       "3    -1 .         \n",
       "4    -1 .         \n",
       "5    -1 .         \n",
       "6    -1 .         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(features_peaks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "## Group by feature, and collapse all peaks associated with that feature\n",
    "  df <- features_peaks %>%\n",
    "    dplyr::mutate(ID = paste(V1, V2, V3, V7)) %>% \n",
    "    dplyr::group_by(ID) %>%\n",
    "    dplyr::reframe(\n",
    "      Peak_ID = paste(unique(V11), collapse = \", \"),\n",
    "      Type = unique(V7),\n",
    "      Divergence = unique(V5) \n",
    "    ) %>%\n",
    "    dplyr::mutate(\n",
    "      Peak = ifelse(Peak_ID == \".\", \"No peak\", \"Peak\")\n",
    "    )\n",
    "\n",
    "  df$Peak <- factor(df$Peak, levels = c(\"Peak\", \"No peak\"))\n",
    "  \n",
    "  print(\n",
    "    ggplot(subset(df, grepl(\"DNA\", Type)), aes(x = Peak, y = Divergence)) +\n",
    "      geom_violin(color = \"grey\", fill = \"grey\") +\n",
    "      geom_boxplot(notch = TRUE, width = .3, outlier.shape = NA) +\n",
    "      theme_minimal() +\n",
    "      theme(axis.title.x = element_text(margin = margin(t = 10)),\n",
    "            axis.text.x = element_text(),\n",
    "            axis.title.y = element_text(margin = margin(r = 15)),\n",
    "            plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "      labs(title = myo_meta$Field_Name[i], \n",
    "            subtitle = \"DNA transposable element divergence\") +\n",
    "      xlab(\"TE status\") + ylab(\"Divergence\") +\n",
    "      scale_y_continuous(labels = comma) \n",
    "  )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "## Plot proportion of peaks in TEs\n",
    "\n",
    "frac_peaks_list <- lapply(seq_along(myo_meta$Abbr), function(i) {\n",
    "  \n",
    "  peak_features <- fread(\n",
    "    file.path(\"results/data\", paste0(myo_meta$Abbr[i], \"_peaks_TEs.tsv\")),\n",
    "    data.table = FALSE\n",
    "  )\n",
    "  peak_features$Type <- peak_features$V11\n",
    "  peak_features$Type[grep(\"^DNA\", peak_features$Type)] <- \"DNA\"\n",
    "  peak_features$Type[grep(\"^SINE\", peak_features$Type)] <- \"SINE\"\n",
    "  peak_features$Type[grep(\"^LINE\", peak_features$Type)] <- \"LINE\"\n",
    "  peak_features$Type[grep(\"^LTR\", peak_features$Type)] <- \"LTR\"\n",
    "  peak_features$Type[grep(\"^Simple\", peak_features$Type)] <- \"Simple repeat\"\n",
    "  peak_features$Type[grep(\"Unknown\", peak_features$Type)] <- \"Unknown\"\n",
    "  peak_features$Type[grep(\"RNA\", peak_features$Type)] <- \"rRNA/tRNA/snRNA\"\n",
    "  peak_features$Type[grep(\"^Sat\", peak_features$Type)] <- \"Other\"\n",
    "  peak_features$Type[grep(\"^Retro\", peak_features$Type)] <- \"Other\"\n",
    "  peak_features$Type[peak_features$Type == \".\"] <- \"NA\"\n",
    "  peak_features$Type <- gsub(\"_\", \" \", peak_features$Type)\n",
    "  \n",
    "  df <- peak_features %>%\n",
    "    dplyr::mutate(TE = ifelse(V10 == \".\", FALSE, TRUE)) %>%\n",
    "    dplyr::group_by(V4) %>%\n",
    "    dplyr::reframe(\n",
    "      TE = unique(TE),\n",
    "      Type = paste(unique(Type), collapse = \", \")\n",
    "    ) %>%\n",
    "    dplyr::mutate(\n",
    "      Type = ifelse(grepl(\",\", Type), \"Unknown\", Type)\n",
    "    ) %>%\n",
    "    dplyr::group_by(TE, Type) %>%\n",
    "    dplyr::reframe(No.Peaks = n()) %>%\n",
    "    dplyr::mutate(Species = myo_meta$Field_Name[i])\n",
    "  \n",
    "  return(df)\n",
    "  \n",
    "})\n",
    "\n",
    "df <- do.call(rbind, frac_peaks_list)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 x 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>TE</th><th scope=col>Type</th><th scope=col>No.Peaks</th><th scope=col>Species</th></tr>\n",
       "\t<tr><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>FALSE</td><td>NA            </td><td>96939</td><td>Myotis-Auriculus</td></tr>\n",
       "\t<tr><td> TRUE</td><td>DNA           </td><td> 2812</td><td>Myotis-Auriculus</td></tr>\n",
       "\t<tr><td> TRUE</td><td>LINE          </td><td>10978</td><td>Myotis-Auriculus</td></tr>\n",
       "\t<tr><td> TRUE</td><td>LTR           </td><td> 3599</td><td>Myotis-Auriculus</td></tr>\n",
       "\t<tr><td> TRUE</td><td>Low complexity</td><td> 3627</td><td>Myotis-Auriculus</td></tr>\n",
       "\t<tr><td> TRUE</td><td>Other         </td><td>   59</td><td>Myotis-Auriculus</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 x 4\n",
       "\\begin{tabular}{llll}\n",
       " TE & Type & No.Peaks & Species\\\\\n",
       " <lgl> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\t FALSE & NA             & 96939 & Myotis-Auriculus\\\\\n",
       "\t  TRUE & DNA            &  2812 & Myotis-Auriculus\\\\\n",
       "\t  TRUE & LINE           & 10978 & Myotis-Auriculus\\\\\n",
       "\t  TRUE & LTR            &  3599 & Myotis-Auriculus\\\\\n",
       "\t  TRUE & Low complexity &  3627 & Myotis-Auriculus\\\\\n",
       "\t  TRUE & Other          &    59 & Myotis-Auriculus\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 x 4\n",
       "\n",
       "| TE &lt;lgl&gt; | Type &lt;chr&gt; | No.Peaks &lt;int&gt; | Species &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| FALSE | NA             | 96939 | Myotis-Auriculus |\n",
       "|  TRUE | DNA            |  2812 | Myotis-Auriculus |\n",
       "|  TRUE | LINE           | 10978 | Myotis-Auriculus |\n",
       "|  TRUE | LTR            |  3599 | Myotis-Auriculus |\n",
       "|  TRUE | Low complexity |  3627 | Myotis-Auriculus |\n",
       "|  TRUE | Other          |    59 | Myotis-Auriculus |\n",
       "\n"
      ],
      "text/plain": [
       "  TE    Type           No.Peaks Species         \n",
       "1 FALSE NA             96939    Myotis-Auriculus\n",
       "2  TRUE DNA             2812    Myotis-Auriculus\n",
       "3  TRUE LINE           10978    Myotis-Auriculus\n",
       "4  TRUE LTR             3599    Myotis-Auriculus\n",
       "5  TRUE Low complexity  3627    Myotis-Auriculus\n",
       "6  TRUE Other             59    Myotis-Auriculus"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<strong>pdf:</strong> 2"
      ],
      "text/latex": [
       "\\textbf{pdf:} 2"
      ],
      "text/markdown": [
       "**pdf:** 2"
      ],
      "text/plain": [
       "pdf \n",
       "  2 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "pdf(\"results/figures/peak_TE_proportion.pdf\", width = 8, height = 6)\n",
    "\n",
    "df <- df %>%\n",
    "  dplyr::group_by(Species) %>%\n",
    "  dplyr::mutate(Frac_Peaks = No.Peaks/sum(No.Peaks))\n",
    "  \n",
    "x_order <- df %>%\n",
    "  dplyr::filter(TE == TRUE) %>%\n",
    "  dplyr::group_by(Species) %>%\n",
    "  dplyr::summarise(n = sum(Frac_Peaks)) %>%\n",
    "  dplyr::arrange(desc(n))\n",
    "  \n",
    "df$Species <- factor(df$Species, levels = x_order$Species)\n",
    "  \n",
    "print(\n",
    "  ggplot(df, aes(x = Species, y = Frac_Peaks, fill = TE, color = TE)) +\n",
    "    geom_bar(stat = \"identity\") +\n",
    "    theme_minimal() +\n",
    "    theme(axis.title.x = element_blank(),\n",
    "          axis.text.x = element_text(angle = 45, hjust = 1),\n",
    "          axis.title.y = element_text(margin = margin(r = 15)),\n",
    "          plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "    labs(title = \"Proportion of peaks in transposable elements\") +\n",
    "    ylab(\"Proportion\") \n",
    ")\n",
    "\n",
    "## Stratify by TE type:\n",
    "\n",
    "df_type <- df %>%\n",
    "  dplyr::filter(TE == TRUE) %>%\n",
    "  dplyr::group_by(Species) %>%\n",
    "  dplyr::mutate(Frac_Peaks = No.Peaks/sum(No.Peaks))\n",
    "  \n",
    "x_order <- df_type %>%\n",
    "  dplyr::group_by(Type) %>%\n",
    "  dplyr::reframe(n = mean(Frac_Peaks)) %>%\n",
    "  dplyr::arrange(desc(n))\n",
    "\n",
    "df_type$Type <- factor(df_type$Type, levels = x_order$Type)\n",
    "\n",
    "print(\n",
    "  ggplot(df_type, aes(x = Type, y = Frac_Peaks)) +\n",
    "    geom_bar(stat = \"identity\") +\n",
    "    theme_minimal() +\n",
    "    theme(axis.title.x = element_blank(),\n",
    "          axis.text.x = element_text(angle = 45, hjust = 1, size = 7),\n",
    "          axis.title.y = element_text(margin = margin(r = 15)),\n",
    "          panel.grid = element_blank(),\n",
    "          plot.margin = unit(c(1, 3, 1, 3), \"cm\")) +\n",
    "    labs(title = \"Proportion of peaks in classes of transposable elements\") +\n",
    "    ylab(\"Proportion\") +\n",
    "    facet_wrap(. ~ Species)\n",
    ")\n",
    "\n",
    "x_order <- df_type %>%\n",
    "  dplyr::group_by(Type) %>%\n",
    "  dplyr::reframe(n = sum(No.Peaks)) %>%\n",
    "  dplyr::arrange(desc(n))\n",
    "\n",
    "df_type$Type <- factor(df_type$Type, levels = x_order$Type)\n",
    "\n",
    "print(\n",
    "  ggplot(df_type, aes(x = Type, y = No.Peaks)) +\n",
    "    geom_bar(stat = \"identity\") +\n",
    "    theme_minimal() +\n",
    "    theme(axis.title.x = element_blank(),\n",
    "          axis.text.x = element_text(angle = 45, hjust = 1, size = 7),\n",
    "          axis.title.y = element_text(margin = margin(r = 15)),\n",
    "          panel.grid = element_blank(),\n",
    "          plot.margin = unit(c(1, 3, 1, 3), \"cm\")) +\n",
    "    labs(title = \"# of peaks in classes of transposable elements\") +\n",
    "    ylab(\"# peaks\") + scale_y_continuous(labels = comma) +\n",
    "    facet_wrap(. ~ Species)\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "## Plot fraction of overall genome comprised of TEs \n",
    "\n",
    "frac_genome_list <- lapply(seq_along(myo_meta$Abbr), function(i) {\n",
    "\n",
    "  ## Load in TE GTF\n",
    "  spec_TEs <- fread(file.path(\"/Users/rebecca/sudmant/analyses/myotis/data/genomes/bat_genomes/repeatMasker\", \n",
    "                              paste0(myo_meta$Abbr[i], \"1_repetMasker.gff3\")), skip = 1, data.table = FALSE)\n",
    "\n",
    "  ## Load in chromosome lengths\n",
    "  chromsize <- fread(paste0(\"/Users/rebecca/sudmant/analyses/myotis/data/\", \n",
    "                            myo_meta$Abbr[i], \"_chromsizes\"), \n",
    "                     data.table = FALSE)\n",
    "\n",
    "  # ## Load in TE peak data\n",
    "  # features_peaks <- fread(\n",
    "  #   file.path(\"results/data\", paste0(myo_meta$Abbr[i], \"_TEs_peaks.tsv\")), \n",
    "  #   data.table = FALSE\n",
    "  # )\n",
    "                          \n",
    "  ## Get fraction of genome occupied by TEs\n",
    "  spec_TEs <- spec_TEs[,c(1, 4:5, 7)] \n",
    "  colnames(spec_TEs) <- c(\"seqnames\", \"start\", \"end\", \"strand\")\n",
    "  bps_TEs <- sum(width(GenomicRanges::reduce(\n",
    "    makeGRangesFromDataFrame(spec_TEs, ignore.strand = TRUE)\n",
    "  )))\n",
    "  frac_genome <- bps_TEs/sum(chromsize[,2])\n",
    "\n",
    "  # ## Get fraction of TEs occupied by peaks:\n",
    "  # features_peaks <- features_peaks %>%\n",
    "  #   dplyr::group_by(ID = paste(V1, V2, V3)) %>% \n",
    "  #   dplyr::reframe(\n",
    "  #     Peak_ID = paste(unique(V11), collapse = \", \")\n",
    "  #   ) %>%\n",
    "  #   dplyr::mutate(\n",
    "  #     Peak = ifelse(Peak_ID == \".\", FALSE, TRUE)\n",
    "  #   )\n",
    "  # frac_peaks <- sum(features_peaks$Peak)/nrow(features_peaks)       \n",
    "\n",
    "  return(data.frame(Species = myo_meta$Field_Name[i], \n",
    "                    Frac_Genome = frac_genome)) # , Frac_Peaks = frac_peaks\n",
    "  \n",
    "}) \n",
    "\n",
    "df <- do.call(rbind, frac_genome_list)\n",
    "df <- df[rev(order(df$Frac_Genome)),]\n",
    "\n",
    "df$Species <- factor(df$Species, levels = df$Species)\n",
    "\n",
    "pdf(\"results/figures/genome_TE_proportion.pdf\", width = 8, height = 8)\n",
    "\n",
    "print(\n",
    "  ggplot(df, aes(x = Species, y = Frac_Genome)) +\n",
    "    geom_bar(stat = \"identity\") +\n",
    "    theme_minimal() +\n",
    "    theme(axis.title.x = element_blank(),\n",
    "          axis.text.x = element_text(angle = 45, hjust = 1),\n",
    "          axis.title.y = element_text(margin = margin(r = 15)),\n",
    "          plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "    labs(title = \"Fraction of genome comprised of transposable elements\") +\n",
    "    ylab(\"Fraction\") +\n",
    "    scale_y_continuous(limits = c(0, .25))\n",
    ")\n",
    "\n",
    "dev.off()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
