{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "setwd(\"/Users/rebecca/sudmant/analyses/myotis/analysis/reciprocal_best_hits/contextualize_best_hits\")\n",
    "\n",
    "library(dplyr, verbose = FALSE)\n",
    "library(tidyr)\n",
    "library(scales)\n",
    "library(ggrepel)\n",
    "library(ggplot2)\n",
    "library(data.table)\n",
    "library(RColorBrewer)\n",
    "\n",
    "myo_meta <- read.csv(\"../../../data/myotis_meta.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# ## Plot summary of peaks in reciprocal regions:\n",
    "\n",
    "# for (i in seq_along(myo_meta$Abbr)) {\n",
    "  \n",
    "#   for (j in seq_along(myo_meta$Abbr)) {\n",
    "\n",
    "#     if (i > j) {\n",
    "\n",
    "#       # pdf(file = paste0(\"results/myo_meta$Abbr[i], \"_vs_\", myo_meta$Abbr[j]_best_hits_peak_summary.pdf\"), width = 7, height = 6)\n",
    "\n",
    "#       peaks1 <- fread(\n",
    "#         file.path(\"resources/genes/peaks\", \n",
    "#         list.files(path = \"resources/genes/peaks\", pattern = paste0(myo_meta$Abbr[i], \"_vs_\", myo_meta$Abbr[j], \".*_peaks.csv\"))),\n",
    "#         data.table = FALSE\n",
    "#         )\n",
    "#       peaks2 <- fread(\n",
    "#         file.path(\"resources/genes/peaks\", \n",
    "#         list.files(path = \"resources/genes/peaks\", pattern = paste0(myo_meta$Abbr[j], \"_vs_\", myo_meta$Abbr[i], \".*_peaks.csv\"))),\n",
    "#         data.table = FALSE\n",
    "#         )\n",
    "\n",
    "#       peaks1$Reciprocal_Seq <- with(peaks1, paste(Chr, Reciprocal_Start, Reciprocal_End))\n",
    "#       peaks2$Reciprocal_Seq <- with(peaks2, paste(Chr, Reciprocal_Start, Reciprocal_End))\n",
    "#       peaks1$Reciprocal_Length <- as.numeric(with(peaks1, Reciprocal_End - Reciprocal_Start))\n",
    "#       peaks2$Reciprocal_Length <- as.numeric(with(peaks2, Reciprocal_End - Reciprocal_Start))\n",
    "#       peaks1$Peak_Length <- as.numeric(with(peaks1, paste(Peak_End - Peak_Start)))\n",
    "#       peaks2$Peak_Length <- as.numeric(with(peaks2, paste(Peak_End - Peak_Start)))\n",
    "\n",
    "#       peaks1_df <- peaks1 %>%\n",
    "#         dplyr::group_by(Chr, Reciprocal_Seq) %>% \n",
    "#         dplyr::summarise(\n",
    "#           No.Peaks = n(),\n",
    "#           No.Peaks_Rel = n()/unique(Reciprocal_Length)\n",
    "#         )\n",
    "\n",
    "#       peaks2_df <- peaks2 %>%\n",
    "#         dplyr::group_by(Chr, Reciprocal_Seq) %>% \n",
    "#         dplyr::summarise(\n",
    "#           No.Peaks = n(),\n",
    "#           No.Peaks_Rel = n()/unique(Reciprocal_Length)\n",
    "#         )\n",
    "\n",
    "#       ## Plot distro of peak lengths:\n",
    "\n",
    "#       print(\n",
    "#         ggplot(peaks1, aes(x = Peak_Length)) + \n",
    "#           geom_histogram(bins = 100) +\n",
    "#           theme_minimal() +\n",
    "#           theme(axis.title.x = element_text(margin = margin(t = 15)),\n",
    "#                 axis.title.y = element_blank()) +\n",
    "#           labs(title = paste(myo_meta$Abbr[i], \"peaks in reciprocal regions with\", myo_meta$Abbr[j])) +\n",
    "#           xlab(\"Peak length (nt)\") +\n",
    "#           scale_x_continuous(labels = comma) +\n",
    "#           scale_y_continuous(labels = comma)\n",
    "#       )\n",
    "\n",
    "#       print(\n",
    "#         ggplot(peaks2, aes(x = Peak_Length)) + \n",
    "#           geom_histogram(bins = 100) +\n",
    "#           theme_minimal() +\n",
    "#           theme(axis.title.x = element_text(margin = margin(t = 15)),\n",
    "#                 axis.title.y = element_blank()) +\n",
    "#           labs(title = paste(myo_meta$Abbr[j], \"peaks in reciprocal regions with\", myo_meta$Abbr[i])) +\n",
    "#           xlab(\"Peak length (nt)\") +\n",
    "#           scale_x_continuous(labels = comma) +\n",
    "#           scale_y_continuous(labels = comma)\n",
    "#       )\n",
    "\n",
    "#       ## Plot distro of # of peaks per reciprocal region:\n",
    "\n",
    "#       print(\n",
    "#         ggplot(peaks1_df, aes(x = No.Peaks)) + \n",
    "#           geom_histogram(bins = 50) +\n",
    "#           theme_minimal() +\n",
    "#           theme(axis.title.x = element_text(margin = margin(t = 15)),\n",
    "#                 axis.title.y = element_blank()) +\n",
    "#           labs(title = paste(myo_meta$Abbr[i], \"peaks in reciprocal regions with\", myo_meta$Abbr[j])) +\n",
    "#           xlab(\"# peaks per region\") +\n",
    "#           scale_x_continuous(labels = comma)\n",
    "#       )\n",
    "\n",
    "#       print(\n",
    "#         ggplot(peaks2_df, aes(x = No.Peaks)) + \n",
    "#           geom_histogram(bins = 50) +\n",
    "#           theme_minimal() +\n",
    "#           theme(axis.title.x = element_text(margin = margin(t = 15)),\n",
    "#                 axis.title.y = element_blank()) +\n",
    "#           labs(title = paste(myo_meta$Abbr[j], \"peaks in reciprocal regions with\", myo_meta$Abbr[i])) +\n",
    "#           xlab(\"# peaks per region\") +\n",
    "#           scale_x_continuous(labels = comma)\n",
    "#       )\n",
    "\n",
    "#       # print(\n",
    "#       #   ggplot(peaks1_df, aes(No.Peaks_Rel)) +\n",
    "#       #     geom_histogram(bins = 30) +\n",
    "#       #     theme_minimal() +\n",
    "#       #     theme(axis.title.x = element_text(margin = margin(t = 15)),\n",
    "#       #           axis.title.y = element_blank()) +\n",
    "#       #     labs(title = paste(myo_meta$Abbr[i], \"peaks in reciprocal regions with\", myo_meta$Abbr[j])) +\n",
    "#       #     xlab(\"# peaks per nucleotide in reciprocal regions\") +\n",
    "#       #     scale_x_continuous(breaks = seq(0, 1, by = .00005))\n",
    "#       # )\n",
    "\n",
    "#       # dev.off()\n",
    "\n",
    "#     }\n",
    "#   }\n",
    "# }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Myotis-Auriculus\"\n",
      "[1] \"Myotis-Californicus\"\n",
      "[1] \"Myotis-Evotis\"\n",
      "[1] \"Myotis-Lucifugus\"\n",
      "[1] \"Myotis-Occultus\"\n",
      "[1] \"Myotis-Thysanodes\"\n",
      "[1] \"Myotis-Velifer\"\n",
      "[1] \"Myotis-Volans\"\n",
      "[1] \"Myotis-Yumanensis\"\n"
     ]
    }
   ],
   "source": [
    "## How do peaks in orthologous genes compare?\n",
    "\n",
    "for (i in seq_along(myo_meta$Abbr)) {\n",
    "\n",
    "  print(myo_meta$Field_Name[i])\n",
    "  \n",
    "  inds <- seq_along(myo_meta$Abbr)\n",
    "\n",
    "  spec2_data <- suppressMessages(lapply(inds[inds != i], function(j) {\n",
    "\n",
    "    ## Load in peak data for species 1: \n",
    "    ## peaks in reciprocal regions that intersect orthologous genes\n",
    "    peaks1_genes <- fread(\n",
    "      file.path(\"resources/genes/peaks\", \n",
    "      list.files(path = \"resources/genes/peaks\", pattern = paste0(myo_meta$Abbr[i], \"_vs_\", myo_meta$Abbr[j], \".*_only.csv\"))),\n",
    "      data.table = FALSE\n",
    "    )\n",
    "    peaks1_genes <- peaks1_genes %>%\n",
    "    dplyr::group_by(myotis_ali_fn) %>%\n",
    "    dplyr::slice_max(as.numeric(Intersect_Start)) %>%\n",
    "    dplyr::select(myotis_ali_fn, Peak_ID)\n",
    "\n",
    "    ## Load in peak data for species 1: \n",
    "    ## Species 1 orthologous genes in reciprocal best hit regions:\n",
    "    hits1_genes <- fread(\n",
    "      file.path(\"resources/genes\", \n",
    "      list.files(path = \"resources/genes\", pattern = paste0(myo_meta$Abbr[i], \"_vs_\", myo_meta$Abbr[j], \".*_only.csv\"))),\n",
    "      data.table = FALSE\n",
    "    )\n",
    "    hits1_genes <- hits1_genes %>%\n",
    "      dplyr::mutate(Reciprocal_Seq = paste(Chr, Reciprocal_Start, Reciprocal_End)) %>%\n",
    "      dplyr::group_by(Chr, myotis_ali_fn, Reciprocal_Seq) %>%\n",
    "      dplyr::summarise(human_gene_name = unique(human_gene_name))\n",
    "\n",
    "    ## Load in peak data for species 2: \n",
    "    ## peaks in reciprocal regions that intersect orthologous genes\n",
    "    peaks2_genes <- fread(\n",
    "      file.path(\"resources/genes/peaks\", \n",
    "      list.files(path = \"resources/genes/peaks\", pattern = paste0(myo_meta$Abbr[j], \"_vs_\", myo_meta$Abbr[i], \".*_only.csv\"))),\n",
    "      data.table = FALSE\n",
    "    )\n",
    "    peaks2_genes <- peaks2_genes %>%\n",
    "      dplyr::group_by(myotis_ali_fn) %>%\n",
    "      dplyr::slice_max(as.numeric(Intersect_Start)) %>%\n",
    "      dplyr::select(Peak_ID)\n",
    "\n",
    "    ## Load in peak data for species 2: \n",
    "    ## Species 1 orthologous genes in reciprocal best hit regions:  \n",
    "    hits2_genes <- fread(\n",
    "      file.path(\"resources/genes\", \n",
    "      list.files(path = \"resources/genes\", pattern = paste0(myo_meta$Abbr[j], \"_vs_\", myo_meta$Abbr[i], \".*_only.csv\"))),\n",
    "      data.table = FALSE\n",
    "    )\n",
    "    hits2_genes <- hits2_genes %>%\n",
    "      dplyr::mutate(Reciprocal_Seq = paste(Chr, Reciprocal_Start, Reciprocal_End)) %>%\n",
    "      dplyr::group_by(Chr, myotis_ali_fn, Reciprocal_Seq) %>%\n",
    "      dplyr::summarise(human_gene_name = unique(human_gene_name))\n",
    "\n",
    "    ## Genes not present in peak files implies no peak at that location:\n",
    "    hits1_status <- merge(hits1_genes, peaks1_genes, by = \"myotis_ali_fn\", all.x = TRUE)\n",
    "    hits1_status$Peak <- TRUE\n",
    "    hits1_status$Peak[is.na(hits1_status$Peak_ID)] <- FALSE\n",
    "    hits2_status <- merge(hits2_genes, peaks2_genes, by = \"myotis_ali_fn\", all.x = TRUE)\n",
    "    hits2_status$Peak <- TRUE\n",
    "    hits2_status$Peak[is.na(hits2_status$Peak_ID)] <- FALSE\n",
    "\n",
    "    ## Subset to orthologous genes present in both species:\n",
    "    hits1_status <- hits1_status[hits1_status$myotis_ali_fn %in% hits2_status$myotis_ali_fn,]\n",
    "    hits2_status <- hits2_status[match(hits1_status$myotis_ali_fn, hits2_status$myotis_ali_fn),]\n",
    "    if (!identical(hits2_status$myotis_ali_fn, hits1_status$myotis_ali_fn)) {\n",
    "      print(paste(i, j))\n",
    "      stop(\"Genes do not match between species\")\n",
    "    }\n",
    "\n",
    "    ## Correlation between peak status in orthologous genes:\n",
    "    peak_cor <- cor(hits1_status$Peak, hits2_status$Peak)\n",
    "    \n",
    "    ## Of the orthologous genes in species 1, what fraction of them have differential peaks to species 2?\n",
    "    is_diff <- hits1_status$Peak != hits2_status$Peak\n",
    "    n_shared_peaks <- as.matrix(table(!is_diff))\n",
    "    \n",
    "    return(\n",
    "      data.frame(\n",
    "        Species = myo_meta$Field_Name[j],\n",
    "        Cor = peak_cor,\n",
    "        Is_Peak_Shared = rownames(n_shared_peaks),\n",
    "        No.Peaks = as.numeric(n_shared_peaks[,1]), \n",
    "        Frac_Peaks = as.numeric(n_shared_peaks[,1])/length(is_diff)\n",
    "      )\n",
    "    ) \n",
    "    \n",
    "  }))\n",
    "  \n",
    "  df <- do.call(rbind, spec2_data)\n",
    "  df$Species <- gsub(\"Myotis-\", \"\", df$Species)\n",
    "  \n",
    "  spec_order <- df %>%\n",
    "    dplyr::filter(Is_Peak_Shared == FALSE) %>%\n",
    "    dplyr::arrange(desc(Frac_Peaks))  \n",
    "\n",
    "  df$Species <- factor(df$Species, levels = spec_order$Species)\n",
    "  df$Is_Peak_Shared <- factor(df$Is_Peak_Shared, levels = c(TRUE, FALSE))\n",
    "\n",
    "  pdf(file = paste0(\"results/\", myo_meta$Abbr[i], \"_orthologous_gene_differential_peaks.pdf\"), width = 7, height = 7)\n",
    "\n",
    "  print(\n",
    "    ggplot(df, aes(x = Species, y = Frac_Peaks, fill = Is_Peak_Shared)) +\n",
    "    geom_bar(stat = \"identity\", color = \"black\") +\n",
    "    theme_minimal() +\n",
    "    theme(plot.subtitle = element_text(margin = margin(b = 10)), \n",
    "          axis.title.x = element_blank(),\n",
    "          axis.text.x = element_text(margin = margin(t = -10, b = 10), size = 11, angle = 45, hjust = 1),\n",
    "          axis.title.y = element_text(margin = margin(r = 15)),\n",
    "          axis.text.y = element_text(size = 11),\n",
    "          axis.ticks.y = element_line(color = \"black\"),\n",
    "          legend.position = \"bottom\",\n",
    "          legend.direction = \"horizontal\",\n",
    "          panel.grid = element_blank(),\n",
    "          plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "    labs(title = myo_meta$Field_Name[i], subtitle = \"Peak status in orthologous genes\") +\n",
    "    ylab(paste(\"Fraction of genes\\n(present in both species)\")) +\n",
    "    scale_fill_manual(name = \"Is peak conserved between species?\", values = brewer.pal(3, \"Paired\"))\n",
    "  ) \n",
    "\n",
    "  df <- df[df$Is_Peak_Shared == FALSE,]\n",
    "\n",
    "  print(\n",
    "    ggplot(df, aes(x = Frac_Peaks, y = Cor)) +\n",
    "    geom_point() +\n",
    "    geom_text_repel(aes(label = Species), size = 4) +\n",
    "    theme_minimal() +\n",
    "    theme(axis.title.x = element_text(size = 12, margin = margin(t = 15)),\n",
    "          axis.text.x = element_text(size = 11),\n",
    "          axis.title.y = element_text(size = 12, margin = margin(r = 15)),\n",
    "          axis.text.y = element_text(size = 11),\n",
    "          axis.ticks.y = element_line(color = \"black\"),\n",
    "          legend.position = \"bottom\",\n",
    "          legend.direction = \"horizontal\",\n",
    "          plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "    labs(title = myo_meta$Field_Name[i], subtitle = \"Peak status in orthologous genes\") +\n",
    "    xlab(\"Fraction of genes with differential peak status\") +\n",
    "    ylab(\"Correlation between peaks status\")\n",
    "  )\n",
    "\n",
    "  dev.off()\n",
    "\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Myotis-Auriculus\"\n",
      "[1] \"Myotis-Californicus\"\n",
      "[1] \"Myotis-Evotis\"\n",
      "[1] \"Myotis-Lucifugus\"\n"
     ]
    }
   ],
   "source": [
    "## How do the genes with differential peaks compare?\n",
    "\n",
    "for (i in seq_along(myo_meta$Abbr)) {\n",
    "\n",
    "  print(myo_meta$Field_Name[i])\n",
    "  \n",
    "  inds <- seq_along(myo_meta$Abbr)\n",
    "\n",
    "  spec2_data <- suppressMessages(lapply(inds[inds != i], function(j) {\n",
    "\n",
    "    ## Load in peak data for species 1: \n",
    "    ## peaks in reciprocal regions that intersect orthologous genes\n",
    "    peaks1_genes <- fread(\n",
    "      file.path(\"resources/genes/peaks\", \n",
    "      list.files(path = \"resources/genes/peaks\", pattern = paste0(myo_meta$Abbr[i], \"_vs_\", myo_meta$Abbr[j], \".*_only.csv\"))),\n",
    "      data.table = FALSE\n",
    "    )\n",
    "    peaks1_genes <- peaks1_genes %>%\n",
    "    dplyr::group_by(myotis_ali_fn) %>%\n",
    "    dplyr::slice_max(as.numeric(Intersect_Start)) %>%\n",
    "    dplyr::select(myotis_ali_fn, Peak_ID)\n",
    "\n",
    "    ## Load in peak data for species 1: \n",
    "    ## Species 1 orthologous genes in reciprocal best hit regions:\n",
    "    hits1_genes <- fread(\n",
    "      file.path(\"resources/genes\", \n",
    "      list.files(path = \"resources/genes\", pattern = paste0(myo_meta$Abbr[i], \"_vs_\", myo_meta$Abbr[j], \".*_only.csv\"))),\n",
    "      data.table = FALSE\n",
    "    )\n",
    "    hits1_genes <- hits1_genes %>%\n",
    "      dplyr::mutate(Reciprocal_Seq = paste(Chr, Reciprocal_Start, Reciprocal_End)) %>%\n",
    "      dplyr::group_by(Chr, myotis_ali_fn, Reciprocal_Seq) %>%\n",
    "      dplyr::summarise(human_gene_name = unique(human_gene_name))\n",
    "\n",
    "    ## Load in peak data for species 2: \n",
    "    ## peaks in reciprocal regions that intersect orthologous genes\n",
    "    peaks2_genes <- fread(\n",
    "      file.path(\"resources/genes/peaks\", \n",
    "      list.files(path = \"resources/genes/peaks\", pattern = paste0(myo_meta$Abbr[j], \"_vs_\", myo_meta$Abbr[i], \".*_only.csv\"))),\n",
    "      data.table = FALSE\n",
    "    )\n",
    "    peaks2_genes <- peaks2_genes %>%\n",
    "      dplyr::group_by(myotis_ali_fn) %>%\n",
    "      dplyr::slice_max(as.numeric(Intersect_Start)) %>%\n",
    "      dplyr::select(Peak_ID)\n",
    "\n",
    "    ## Load in peak data for species 2: \n",
    "    ## Species 1 orthologous genes in reciprocal best hit regions:  \n",
    "    hits2_genes <- fread(\n",
    "      file.path(\"resources/genes\", \n",
    "      list.files(path = \"resources/genes\", pattern = paste0(myo_meta$Abbr[j], \"_vs_\", myo_meta$Abbr[i], \".*_only.csv\"))),\n",
    "      data.table = FALSE\n",
    "    )\n",
    "    hits2_genes <- hits2_genes %>%\n",
    "      dplyr::mutate(Reciprocal_Seq = paste(Chr, Reciprocal_Start, Reciprocal_End)) %>%\n",
    "      dplyr::group_by(Chr, myotis_ali_fn, Reciprocal_Seq) %>%\n",
    "      dplyr::summarise(human_gene_name = unique(human_gene_name))\n",
    "\n",
    "    ## Genes not present in peak files implies no peak at that location:\n",
    "    hits1_status <- merge(hits1_genes, peaks1_genes, by = \"myotis_ali_fn\", all.x = TRUE)\n",
    "    hits1_status$Peak <- TRUE\n",
    "    hits1_status$Peak[is.na(hits1_status$Peak_ID)] <- FALSE\n",
    "    hits2_status <- merge(hits2_genes, peaks2_genes, by = \"myotis_ali_fn\", all.x = TRUE)\n",
    "    hits2_status$Peak <- TRUE\n",
    "    hits2_status$Peak[is.na(hits2_status$Peak_ID)] <- FALSE\n",
    "\n",
    "    ## Subset to orthologous genes present in both species:\n",
    "    hits1_status <- hits1_status[hits1_status$myotis_ali_fn %in% hits2_status$myotis_ali_fn,]\n",
    "    hits2_status <- hits2_status[match(hits1_status$myotis_ali_fn, hits2_status$myotis_ali_fn),]\n",
    "    if (!identical(hits2_status$myotis_ali_fn, hits1_status$myotis_ali_fn)) {\n",
    "      print(paste(i, j))\n",
    "      stop(\"Genes do not match between species\")\n",
    "    }\n",
    "\n",
    "    ## Correlation between peak status in orthologous genes:\n",
    "    peak_cor <- cor(hits1_status$Peak, hits2_status$Peak)\n",
    "    \n",
    "    ## Of the orthologous genes in species 1, what fraction of them have differential peaks to species 2?\n",
    "    is_diff <- hits1_status$Peak != hits2_status$Peak\n",
    "    n_shared_peaks <- as.matrix(table(!is_diff))\n",
    "    \n",
    "    return(\n",
    "      data.frame(\n",
    "        Species = myo_meta$Field_Name[j],\n",
    "        Cor = peak_cor,\n",
    "        Is_Peak_Shared = rownames(n_shared_peaks),\n",
    "        No.Peaks = as.numeric(n_shared_peaks[,1]), \n",
    "        Frac_Peaks = as.numeric(n_shared_peaks[,1])/length(is_diff)\n",
    "      )\n",
    "    ) \n",
    "    \n",
    "  }))\n",
    "  \n",
    "  df <- do.call(rbind, spec2_data)\n",
    "  df$Species <- gsub(\"Myotis-\", \"\", df$Species)\n",
    "  df$Is_Peak_Shared <- factor(df$Is_Peak_Shared, levels = c(TRUE, FALSE))\n",
    "\n",
    "  pdf(file = paste0(\"results/\", myo_meta$Abbr[i], \"_orthologous_gene_differential_peaks.pdf\"), width = 7, height = 7)\n",
    "\n",
    "  spec_order <- df %>%\n",
    "    dplyr::filter(Is_Peak_Shared == FALSE) %>%\n",
    "    dplyr::arrange(desc(Frac_Peaks))  \n",
    "\n",
    "  df$Species <- factor(df$Species, levels = spec_order$Species)\n",
    "\n",
    "  print(\n",
    "    ggplot(df, aes(x = Species, y = Frac_Peaks, fill = Is_Peak_Shared)) +\n",
    "    geom_bar(stat = \"identity\", color = \"black\") +\n",
    "    theme_minimal() +\n",
    "    theme(plot.subtitle = element_text(margin = margin(b = 10)), \n",
    "          axis.title.x = element_blank(),\n",
    "          axis.text.x = element_text(margin = margin(t = -10, b = 10), size = 11, angle = 45, hjust = 1),\n",
    "          axis.title.y = element_text(margin = margin(r = 15)),\n",
    "          axis.text.y = element_text(size = 11),\n",
    "          axis.ticks.y = element_line(color = \"black\"),\n",
    "          legend.position = \"bottom\",\n",
    "          legend.direction = \"horizontal\",\n",
    "          panel.grid = element_blank(),\n",
    "          plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "    labs(title = myo_meta$Field_Name[i], subtitle = \"Peak status in orthologous genes\") +\n",
    "    ylab(paste(\"Fraction of genes\\n(present in both species)\")) +\n",
    "    scale_fill_manual(name = \"Is peak conserved between species?\", values = brewer.pal(3, \"Paired\"))\n",
    "  )\n",
    "  \n",
    "  spec_order <- df %>%\n",
    "    dplyr::filter(Is_Peak_Shared == FALSE) %>%\n",
    "    dplyr::arrange(desc(No.Peaks))  \n",
    "\n",
    "  df$Species <- factor(df$Species, levels = spec_order$Species)\n",
    "\n",
    "  print(\n",
    "    ggplot(df, aes(x = Species, y = No.Peaks, fill = Is_Peak_Shared)) +\n",
    "    geom_bar(stat = \"identity\", position = \"dodge\", color = \"black\") +\n",
    "    theme_minimal() +\n",
    "    theme(plot.subtitle = element_text(margin = margin(b = 10)), \n",
    "          axis.title.x = element_blank(),\n",
    "          axis.text.x = element_text(margin = margin(t = -10, b = 10), size = 11, angle = 45, hjust = 1),\n",
    "          axis.title.y = element_text(margin = margin(r = 15)),\n",
    "          axis.text.y = element_text(size = 11),\n",
    "          axis.ticks.y = element_line(color = \"black\"),\n",
    "          legend.position = \"bottom\",\n",
    "          legend.direction = \"horizontal\",\n",
    "          panel.grid = element_blank(),\n",
    "          plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "    labs(title = myo_meta$Field_Name[i], subtitle = \"Peak status in orthologous genes\") +\n",
    "    ylab(paste(\"Fraction of genes\\n(present in both species)\")) +\n",
    "    scale_y_continuous(labels = comma) +\n",
    "    scale_fill_manual(name = \"Is peak conserved between species?\", values = brewer.pal(3, \"Paired\"))\n",
    "  )  \n",
    "\n",
    "  df <- df[df$Is_Peak_Shared == FALSE,]\n",
    "\n",
    "  print(\n",
    "    ggplot(df, aes(x = Frac_Peaks, y = Cor)) +\n",
    "    geom_point() +\n",
    "    geom_text_repel(aes(label = Species), size = 4) +\n",
    "    theme_minimal() +\n",
    "    theme(axis.title.x = element_text(size = 12, margin = margin(t = 15)),\n",
    "          axis.text.x = element_text(size = 11),\n",
    "          axis.title.y = element_text(size = 12, margin = margin(r = 15)),\n",
    "          axis.text.y = element_text(size = 11),\n",
    "          axis.ticks.y = element_line(color = \"black\"),\n",
    "          legend.position = \"bottom\",\n",
    "          legend.direction = \"horizontal\",\n",
    "          plot.margin = unit(c(1, 1, 1, 1), \"cm\")) +\n",
    "    labs(title = myo_meta$Field_Name[i], subtitle = \"Peak status in orthologous genes\") +\n",
    "    xlab(\"Fraction of genes with differential peak status\") +\n",
    "    ylab(\"Correlation between peaks status\")\n",
    "  )\n",
    "\n",
    "  dev.off()\n",
    "\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 x 5</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Species</th><th scope=col>Cor</th><th scope=col>Is_Peak_Shared</th><th scope=col>No.Peaks</th><th scope=col>Frac_Peaks</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>Auriculus   </td><td>0.4482415</td><td>FALSE</td><td>750</td><td>0.1631499</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>Californicus</td><td>0.5845749</td><td>FALSE</td><td>243</td><td>0.1110096</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>Evotis      </td><td>0.3926914</td><td>FALSE</td><td>831</td><td>0.1777160</td></tr>\n",
       "\t<tr><th scope=row>7</th><td>Lucifugus   </td><td>0.4150351</td><td>FALSE</td><td>648</td><td>0.1356216</td></tr>\n",
       "\t<tr><th scope=row>9</th><td>Occultus    </td><td>0.4427453</td><td>FALSE</td><td>667</td><td>0.1626036</td></tr>\n",
       "\t<tr><th scope=row>11</th><td>Thysanodes  </td><td>0.3047411</td><td>FALSE</td><td>816</td><td>0.1999510</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 x 5\n",
       "\\begin{tabular}{r|lllll}\n",
       "  & Species & Cor & Is\\_Peak\\_Shared & No.Peaks & Frac\\_Peaks\\\\\n",
       "  & <fct> & <dbl> & <fct> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & Auriculus    & 0.4482415 & FALSE & 750 & 0.1631499\\\\\n",
       "\t3 & Californicus & 0.5845749 & FALSE & 243 & 0.1110096\\\\\n",
       "\t5 & Evotis       & 0.3926914 & FALSE & 831 & 0.1777160\\\\\n",
       "\t7 & Lucifugus    & 0.4150351 & FALSE & 648 & 0.1356216\\\\\n",
       "\t9 & Occultus     & 0.4427453 & FALSE & 667 & 0.1626036\\\\\n",
       "\t11 & Thysanodes   & 0.3047411 & FALSE & 816 & 0.1999510\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 x 5\n",
       "\n",
       "| <!--/--> | Species &lt;fct&gt; | Cor &lt;dbl&gt; | Is_Peak_Shared &lt;fct&gt; | No.Peaks &lt;dbl&gt; | Frac_Peaks &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|\n",
       "| 1 | Auriculus    | 0.4482415 | FALSE | 750 | 0.1631499 |\n",
       "| 3 | Californicus | 0.5845749 | FALSE | 243 | 0.1110096 |\n",
       "| 5 | Evotis       | 0.3926914 | FALSE | 831 | 0.1777160 |\n",
       "| 7 | Lucifugus    | 0.4150351 | FALSE | 648 | 0.1356216 |\n",
       "| 9 | Occultus     | 0.4427453 | FALSE | 667 | 0.1626036 |\n",
       "| 11 | Thysanodes   | 0.3047411 | FALSE | 816 | 0.1999510 |\n",
       "\n"
      ],
      "text/plain": [
       "   Species      Cor       Is_Peak_Shared No.Peaks Frac_Peaks\n",
       "1  Auriculus    0.4482415 FALSE          750      0.1631499 \n",
       "3  Californicus 0.5845749 FALSE          243      0.1110096 \n",
       "5  Evotis       0.3926914 FALSE          831      0.1777160 \n",
       "7  Lucifugus    0.4150351 FALSE          648      0.1356216 \n",
       "9  Occultus     0.4427453 FALSE          667      0.1626036 \n",
       "11 Thysanodes   0.3047411 FALSE          816      0.1999510 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
